src/test/scala/com/yotpo/metorikku/code/steps/functions/test/UserDefinedFunctionsTests.scala

```diff
@@ -0,0 +1,18 @@
+package com.yotpo.metorikku.code.steps.functions.test
+
+import java.sql.Timestamp
+
+import org.scalatest.{FunSuite, Suites}
+import com.yotpo.metorikku.code.steps.functions.UserDefinedFunctions._
+
+class SerDefinedFunctionsTests extends Suites (
+  new EpochMilliToTimestampTest
+)
+
+class EpochMilliToTimestampTest extends FunSuite {
+
+  test("Given a Long representing Epoch Milli returns a Timestamp") {
+    assert(epochMilliToTimestamp(1584176754000L) === new Timestamp(1584176754000L))
```

Comment:
@Doe-Ed sorry, I'm not sure I understand why the assert it working. I thought it should be - 
assert(epochMilliToTimestamp(1584176754000L) === new Timestamp(1584176754L))
Can you explain?

Reply:
ofEpoch floorDiv by 1000
from (Timestamp) multiplies by 1000

public static Instant ofEpochMilli(long epochMilli) {
  long secs = Math.floorDiv(epochMilli, 1000);
  int mos = (int)Math.floorMod(epochMilli, 1000);
  return create(secs, mos * 1000_000);
}

public static Timestamp from(Instant instant) {
  try {
    Timestamp stamp = new Timestamp(instant.getEpochSecond() * MILLIS_PER_SECOND);
    stamp.nanos = instant.getNano();
    return stamp;
  } catch (ArithmeticException ex) {
    throw new IllegalArgumentException(ex);
  }
}

Reply:
MILLIS_PER_SECOND == 1000